src/main/java/com/aws/greengrass/clientdevices/auth/api/usecases/UseCaseDecorator.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.api.usecases;
+
+import com.aws.greengrass.logging.api.Logger;
+import com.aws.greengrass.logging.impl.LogManager;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+class UseCaseDecorator implements InvocationHandler {
```

Comment:
Why this? - I thought it might be a good idea to have some sort of logging about useCase runtime to verify if introducing useCases is impacting application run time in a negative way. Additionally this will give us a bit more insight into what the application is doing and potentially in the future this are metrics we could use and verify if we are meeting certain SLAs